libafl_qemu/src/snapshot.rs

```diff
@@ -681,32 +681,33 @@ where
             // TODO handle huge pages
 
             #[cfg(any(cpu_target = "arm", cpu_target = "mips"))]
-            if i64::from(sys_num) == SYS_mmap2 {
+            if sys_const == SYS_mmap2 {
                 if let Ok(prot) = MmapPerms::try_from(a2 as i32) {
                     let h = hooks.match_helper_mut::<QemuSnapshotHelper>().unwrap();
                     h.add_mapped(result as GuestAddr, a1 as usize, Some(prot));
                 }
             }
 
             #[cfg(not(cpu_target = "arm"))]
-            if i64::from(sys_num) == SYS_mmap {
+            if sys_const == SYS_mmap {
                 if let Ok(prot) = MmapPerms::try_from(a2 as i32) {
                     let h = hooks.match_helper_mut::<QemuSnapshotHelper>().unwrap();
                     h.add_mapped(result as GuestAddr, a1 as usize, Some(prot));
                 }
             }
 
-            if i64::from(sys_num) == SYS_mremap {
+            if sys_const == SYS_mremap {
                 let h = hooks.match_helper_mut::<QemuSnapshotHelper>().unwrap();
-                h.remove_mapped(a0 as GuestAddr, a1 as usize);
-                h.add_mapped(result as GuestAddr, a2 as usize, None);
-                // TODO get the old permissions from the removed mappin
-            } else if i64::from(sys_num) == SYS_mprotect {
+                if let Ok(prot) = MmapPerms::try_from(a3 as i32) {
```

Comment:
Looking at this now, I m not sure this is correct here. like mixing oranges and apples kind of thing.